Patch migrated from Desktop, where it was listed as:
# date:2010-10-20 owner:yippi type:bug bugster:6993687

aka:
15676303 SUNBT6993687 Solaris D-Bus needs logic to recognize the "console user"

Solaris-specific

--- dbus-1.2.24/dbus/dbus-sysdeps-util-unix.c-orig	2010-10-19 13:27:19.672465743 -0500
+++ dbus-1.2.24/dbus/dbus-sysdeps-util-unix.c	2010-10-19 13:28:39.691609495 -0500
@@ -65,6 +65,10 @@
 #define O_BINARY 0
 #endif
 
+#ifdef __sun
+#include <pwd.h>
+#endif
+
 /**
  * @addtogroup DBusInternalsUtils
  * @{
--- a/dbus/dbus-userdb-util.c	2025-01-15 08:58:47.349300651 -0800
+++ b/dbus/dbus-userdb-util.c	2025-01-15 20:09:03.623855420 -0800
@@ -94,6 +94,18 @@
   DBusString f;
   DBusStat st;
 
+#ifdef __sun
+  if (stat ("/dev/vt/console_user", &st) == 0 && st.st_uid == uid)
+    {
+         /*
+          * Owner is allowed to take over. Before we have real
+          * ownership in HAL, assume it's the console owner.
+          */
+          return TRUE;
+    }
+#else
+
+
   if (!_dbus_string_init (&f))
     {
       _DBUS_SET_OOM (error);
@@ -114,6 +126,7 @@
     }
 
   _dbus_string_free(&f);
+#endif
 
 #endif /* HAVE_CONSOLE_OWNER_FILE */
 
